model/src/main/java/uk/gov/hmcts/payment/api/service/UserAwareDelegatingCardPaymentService.java

```diff
@@ -128,9 +119,23 @@ public PaymentFeeLink retrieve(String paymentReference) {
 
         PaymentFeeLink paymentFeeLink = payment.getPaymentLink();
 
-        String paymentTargetService = payment.getServiceType();
+        String paymentService = payment.getServiceType();
 
-        GovPayPayment govPayPayment = delegate.retrieve(payment.getExternalReference(), paymentTargetService);
+        if (null == paymentService || paymentService.trim().equals("")) {
+            LOG.error("Unable to determine the payment service which created this payment-Ref:" + paymentReference);
+        }
+
+        String callingService = null;
+
+        try {
+            callingService = serviceIdSupplier.get();
+        } catch (NullPointerException exp) {
+            //where some endpoints are not using S2S, in those scenarios it should continue as is.
```

Comment:
it's a bit odd to catch NullPointerException. Can we change the method to return null or Optional<String>?